cmd/xl-v1-multipart.go

```diff
@@ -727,16 +722,12 @@ func (xl xlObjects) PutObjectPart(bucket, object, uploadID string, partID int, s
 	// Add the current part.
 	xlMeta.AddObjectPart(partID, partSuffix, newMD5Hex, size)
 
-	for index, disk := range onlineDisks {
-		if disk == nil {
-			continue
+	for i, disk := range onlineDisks {
+		if disk != OfflineDisk {
```

Comment:
Yes, I'm aware of the way Go handles the happy path (and I like this style by the way):
> Handle errors first, keep the happy path to the left...

What I'm saying is:
If we have just one condition we may can use:
```
if !cond {
}
```
But now I think we should do it always the same way...
So:
```
if cond { return err / continue}

if cond2 { return err / continue}

// happy path
```